source/cursorManager.py

```diff
@@ -264,12 +264,27 @@ def script_moveBySentence_forward(self,gesture):
 		self._caretMovementScriptHelper(gesture,textInfos.UNIT_SENTENCE,1)
 	script_moveBySentence_forward.resumeSayAllMode = sayAll.CURSOR.CARET
 
+	def _handleParagraphNavigation(self, gesture, nextParagraph: bool):
+		setting = config.conf["paragraphNavigation"]["paragraphStyle"]
+		if setting == "block":
+			from utils.paragraphHelper import moveToBlockParagraph
+			ti = self.makeTextInfo(textInfos.POSITION_SELECTION)
+			passKey, moved = moveToBlockParagraph(
+				nextParagraph=nextParagraph, speakNew=not willSayAllResume(gesture), ti=ti)
+			if moved:
+				self.selection = ti
+			elif passKey:
+				# fail over to default behavior
+				self._caretMovementScriptHelper(gesture, textInfos.UNIT_PARAGRAPH, 1 if nextParagraph else -1)
+		else:

```

Comment:
Explicitly check other types:

```suggestion
		elif:
			...
		else:
			log.error(f"Unexpected setting value {value}")
```